### DISCUSSION TITLE
how to get value from _winrt._winrt_base object

### DISCUSSION THREAD

**Discussion by kfal** [pywinrt/python-winsdk#22] [Q&A] [answered]
```python
import winrt.windows.devices.sensors as ds

als = ds.LightSensor.get_default()
lux = als.get_current_reading()
x = lux.properties.lookup("{C458F8A7-4AE8-4777-9607-2E9BDD65110A} 62")
print(x) ---> **problem , <_winrt._winrt_base object at 0x0000023045BC4D10> ,  not able to obtain value**
```

From **https://github.com/tpn/winsdk-10/blob/master/Include/10.0.16299.0/shared/sensorsdef.h**
lux.illuminance_in_lux is lux.properties.lookup("{C458F8A7-4AE8-4777-9607-2E9BDD65110A} 60")
but we can get float at lux.illuminance_in_lux

**Comment by dlech** [comment, marked as answer]
Python tip: you can use Python subscript expression instead of calling `lookup` on a IMapView:

```python
lux.properties["{C458F8A7-4AE8-4777-9607-2E9BDD65110A} 62"]
```

Since this returns a generic WinRT object type (equivalent of .NET System.Object), you probably need to use the `IPropertyValue` APIs to get the value. See https://github.com/pywinrt/pywinrt/blob/main/test/test_property_value.py for extensive examples. 

Maybe something like this?

```python
import winsdk.windows.devices.sensors as ds
import winsdk.windows.foundation as wf

als = ds.LightSensor.get_default()
lux = als.get_current_reading()
p_x = lux.properties["{C458F8A7-4AE8-4777-9607-2E9BDD65110A} 62"]
x = wf.IPropertyValue._from(p_x).get_single()
print(x)
```

Also see https://github.com/pywinrt/pywinrt/issues/8.

**Comment by kfal** [comment]
it works , thanks David
